spec/supabase_kt_v0.yml

```diff
@@ -0,0 +1,3502 @@
+          ) {
+              install(Postgrest) {
+                  defaultSchema = "schema" // default: "public"
+                  propertyConversionMethod = PropertyConversionMethod.SERIAL_NAME // default: PropertyConversionMethod.CAMEL_CASE_TO_SNAKE_CASE
+              }
+          }
+          ```
+        description: |
+          `propertyConversionMethod` - The method to use to convert the property names to the column names when applying filters & using the update method. Default: `PropertyConversionMethod.CAMEL_CASE_TO_SNAKE_CASE`
+          
+          `defaultSchema` - The default schema to use for database requests. Default: `public`
+          
+          `customUrl` - Custom url for the Postgrest Api. Can be safely ignored when using supabase. Default: `null`
+          
+          `jwtToken` - Plugin specific JWT Token. Can be ignored when using the GoTrue plugin. Default: `null`
+      - id: configure-storage
+        name: Configure Storage module
+        code: |
+          ```kotlin
+          val client = createSupabaseClient(
+              supabaseUrl = "https://xyzcompany.supabase.co", 
+              supabaseKey = "public-anon-key"
+          ) {
+              install(Storage) {
+                  transferTimeout = 90.seconds //default: 120 seconds
+              }
+          }
+          ```
+        description: |
+          `transferTimeout` - the timeout for uploading and downloading files. Default: `120.seconds`
+          
+          `resumable.cache` - Interface for storing resumable upload urls. Default: `ResumableCache.Disk`
+          
+          `resumable.defaultChunkSize` - The default chunk size for resumable uploads. Supabase currently only supports a chunk size of 6MB, so be careful when changing this value. Default: `6MB`
+          
+          `resumable.retryTimeout` - the timeout for retrying resumable uploads when uploading a chunk fails. Default: `5.seconds`
+          
+          `resumable.onlyUpdateStateAfterChunk` - whether the upload state should only be updated after a chunk was uploaded successfully or also when the chunk is currently being uploaded. Default: `false`
+          
+          `customUrl` - Custom url for the Storage Api. Can be safely ignored when using supabase. Default: `null`
+          
+          `jwtToken` - Plugin specific JWT Token. Can be ignored when using the GoTrue plugin. Default: `null`
+      - id: configure-realtime
+        name: Configure Realtime module
+        code: |
+          ```kotlin
+          val client = createSupabaseClient(
+              supabaseUrl = "https://xyzcompany.supabase.co", 
+              supabaseKey = "public-anon-key"
+          ) {
+              install(Realtime) {
+                  reconnectDelay = 5.seconds //Default: 7 seconds
+              }
+          }
+          ```
+        description: |
+          `reconnectDelay` - The delay between reconnect attempts. Default: `7 seconds`
```

Comment:
Format of the time seems a bit inconsistent, sometimes it is "x.seconds", sometimes it is "x seconds"